syntax = "proto3";

package key_value;

// A simple key-value storage service.
service KeyValueStore {
  // Look up the value associated with a given key.
  rpc GetValue(GetRequest) returns (GetResponse) {}

  // Look up the value associated with a given key.
  rpc StoreValue(StoreRequest) returns (GetResponse) {}
}

// A request for the value associated with a given key.
message GetRequest {
  string key = 1;
}


// A key-value pair.
message KeyValuePair {
  // The key identifying the pair.
  string key = 1;

  // The string value of the pari.
  string value = 2;
}

message StoreRequest {
  KeyValuePair key_value_pair = 1;
}

message GetResponse {
  KeyValuePair key_value_pair = 1;
}

message StoreResponse {
  KeyValuePair key_value_pair = 1;
}
